### DISCUSSION TITLE
Add "has verified email" eloquent scope for the MustVerifyEmail trait / contract

### DISCUSSION THREAD

**Discussion by angelej** [laravel/framework#51003] [Ideas]
Hi,
I'm curious if this feature would be beneficial for other developers.
```php
// User model implements MustVerifyEmail contract
User::hasVerifiedEmail()->get();

User::hasUnverifiedEmail()->get();
// or
User::hasVerifiedEmail(false)->get();
```
This feature could be implemented in laravel 12 because we have to extend the `Illuminate\Contracts\Auth\MustVerifyEmail` interface. However, we have to come up with a different scope name, since the `MustVerifyEmail` trait and contract already have a method called `hasVerifiedEmail`.

Let me know if you would like to see this feature in the framework and which name we should pick!

**Comment by Kamandlou** [comment]
I like this feature let me know if you need help implementing this feature :))
